src/app/components/game/game.component.ts

```diff
@@ -819,4 +866,46 @@ export class GameComponent implements OnInit, OnChanges {
             }, milliDuration);
         }, milliDuration);
     }
+
+    private drawArrows(): void {
+        this.boardContext.globalAlpha = 0.5;
+        for (const arrow of this.drawnArrows) {
+            const fromX = arrow.fromSquare.file * 80 + 40;
+            const fromY = arrow.fromSquare.rank * 80 + 40;
+            const toX = arrow.toSquare.file * 80 + 40;
+            const toY = arrow.toSquare.rank * 80 + 40;
+            const color = arrow.color;
+            // pointer displacement from the center
+            const displacement = arrow.displacement;
+            // radius of the circumcircle of the triangle pointer
+            const r = arrow.pointerSize;
+            const lineWidth = arrow.lineWidth;
+
+            // imaginary triangle - hypotenuse goes from origin sq to dest sq
+            const angle = Math.atan2(toY - fromY, toX - fromX);
+            const hypotenuse =
+                Math.sqrt((toX - fromX) ** 2 + (toY - fromY) ** 2) +
+                displacement;
+            this.boardContext.save();
+            this.boardContext.translate(fromX, fromY);
+            this.boardContext.rotate(angle);
+
+            // draws the line
+            this.boardContext.beginPath();
+            this.boardContext.moveTo(0, 0);
+            this.boardContext.lineTo(hypotenuse - r, 0);
+            this.boardContext.lineWidth = lineWidth;
+            this.boardContext.strokeStyle = color;
+            this.boardContext.stroke();
+
+            // draws the triangle pointer
+            this.boardContext.beginPath();
+            this.boardContext.lineTo(hypotenuse - r, r);
+            this.boardContext.lineTo(hypotenuse, 0);
+            this.boardContext.lineTo(hypotenuse - r, -r);
+            this.boardContext.fillStyle = color;
+            this.boardContext.fill();
+            this.boardContext.restore();
```

Comment:
Can we get clarity on why `restore` is part of this last step? Or is it separate and not related to the last step? (its own comment?) just doesn't look right.

Reply:
Because we're changing the coordinates in `translate()`. So we have to save the state of the canvas before that and then restore it. Try removing `restore()` -- it'll be fun!